### PR TITLE
[graphql] retire tx digest scalar type

### PR DESCRIPTION
**Pull request by wlmyng**
## Description 
In https://github.com/MystenLabs/sui/pull/13718 there was a request to remove the dedicated scalar for TransactionDigest, and instead return digests as String based on the graphql schema and to limit use of custom scalars. This PR addresses the request, but keeps the Digest type around to minimize String conversions. 

## Test Plan 

snapshot + some manual queries

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/src/types/digest.rs
Comment (amnn) [inline crates/sui-graphql-rpc/src/types/digest.rs:1]: Is it possible to preserve the tests that were in `tx_digest.rs` over here?

File: crates/sui-graphql-rpc/src/context_data/sui_sdk_data_provider.rs [hunk 1]
```diff
@@ -314,18 +311,6 @@ impl DataProvider for SuiClient {
         Ok(connection)
     }
 
-    async fn fetch_tx(&self, digest: &str) -> Result<Option<TransactionBlock>> {
```
Comment (amnn) [inline crates/sui-graphql-rpc/src/context_data/sui_sdk_data_provider.rs:317]: Out of interest, how come this PR removes this function?

**Review by amnn — APPROVED** [review]
Change looks good to me, thanks for the clean-up @wlmyng !  Accepting to unblock, but could you look at preserving the tests (see inline comment)?

File: crates/sui-graphql-rpc/src/context_data/sui_sdk_data_provider.rs (hunk 1 above)
Comment (wlmyng) [reply crates/sui-graphql-rpc/src/context_data/sui_sdk_data_provider.rs:317]: @oxade implemented a DataLoader for transactions and replaced fetch_tx usage, but I suppose that doesn't mean we need to remove this fetch_tx fn

File: crates/sui-graphql-rpc/src/types/digest.rs
Comment (wlmyng) [reply crates/sui-graphql-rpc/src/types/digest.rs:1]: Yes I can bring some back. They mostly have to do with ScalarType and InputType conversions, which aren't that applicable. But I can bring back the first test, and add another one that validates we return an error on invalid Base58 strings

**Pull request merged**